core/src/main/java/de/evoila/cf/broker/service/impl/DeploymentServiceImpl.java

```diff
@@ -158,17 +158,16 @@ public ServiceInstanceOperationResponse updateServiceInstance(String serviceInst
     public ServiceInstanceOperationResponse deleteServiceInstance(String instanceId)
             throws ServiceBrokerException, ServiceInstanceDoesNotExistException, ServiceDefinitionDoesNotExistException {
         ServiceInstance serviceInstance;
+        Plan plan;
         try {
             serviceInstance = serviceInstanceRepository.getServiceInstance(instanceId);
+            plan = serviceDefinitionRepository.getPlan(serviceInstance.getPlanId());
```

Comment:
This method will throw a ServiceDefinitionDoesNotExistException, if no plan was found for the given plan id.
The following catch block will always throw a ServiceInstanceDoesNotExistException, which is not true in that case.